FluidNC/src/Maslow/Maslow.cpp

```diff
@@ -330,17 +303,100 @@ void Maslow_::comply(){
     axisBR.reset();
     log_info("Complying All");
 }
-//updating encoder positions for all 4 arms
+
+//updating encoder positions for all 4 arms, cycling through them each call, at ENCODER_READ_FREQUENCY_HZ frequency
 bool Maslow_::updateEncoderPositions(){
     bool success = true;
-    if(!readingFromSD){
-        if(!axisBL.updateEncoderPosition()) success = false;
-        if(!axisBR.updateEncoderPosition()) success = false;
-        if(!axisTR.updateEncoderPosition()) success = false;
-        if(!axisTL.updateEncoderPosition()) success = false;
+    static unsigned long lastCallToEncoderRead = millis();
+
+    static int encoderFailCounter[4] = {0,0,0,0}; 
+    static unsigned long encoderFailTimer = millis();
+
+    if(!readingFromSD  && (millis() - lastCallToEncoderRead > 1000/(ENCODER_READ_FREQUENCY_HZ) ) ) {
+        static int encoderToRead = 0;
+        switch(encoderToRead){
+            case 0:
+                if(!axisTL.updateEncoderPosition()){
+                    encoderFailCounter[0]++;
+                }
+                break;
+            case 1:
+                if(!axisTR.updateEncoderPosition()){
+                    encoderFailCounter[1]++;
+                }
+                break;
+            case 2:
+                if(!axisBL.updateEncoderPosition()){
+                    encoderFailCounter[2]++;
+                }
+                break;
+            case 3:
+                if(!axisBR.updateEncoderPosition()){
+                    encoderFailCounter[3]++;
+                }
+                break;
+        }
+        encoderToRead++;
+        if(encoderToRead > 3) {
+            encoderToRead = 0;
+            lastCallToEncoderRead = millis();
+        }
     }
+
+    // if more than 10% of readings fail, warn user, if more than 50% fail, stop the machine and raise alarm
```

Comment:
10% seems crazy high. I think that we should be warning on any encoder fails...especially right now because if something is causing them to fail we want to know about it.

Reply:
yeah, the logic was at 10% failure it would still run files perfectly fine

Reply:
Actually, I'm wrong, because call to i2c has it's own 50ms timeout, so 10 failed readings means 0.5 sec of failed readings

Reply:
I kinda like how it used to work where it would print every read failure but then it would start a count down and not print again for one second (that way it won't spam the serial port if one of the encoders is unplugged).

Are you getting encoder read failures? I absolutely never get them unless one of the encoders is unplugged